src/main/java/com/wavefront/sdk/common/ReconnectingSocket.java

```diff
@@ -55,13 +54,20 @@
   public ReconnectingSocket(String host, int port, SocketFactory socketFactory,
                             WavefrontSdkMetricsRegistry sdkMetricsRegistry, String entityPrefix)
       throws IOException {
-    this.host = host;
-    this.port = port;
+    this(new InetSocketAddress(host, port), socketFactory, sdkMetricsRegistry, entityPrefix);
+  }
+
+  /**
+   * @throws IOException When we cannot open the remote socket.
+   */
+  public ReconnectingSocket(InetSocketAddress address, SocketFactory socketFactory,
+                            WavefrontSdkMetricsRegistry sdkMetricsRegistry, String entityPrefix)
+      throws IOException {
+    this.address = address;
     this.serverTerminated = false;
     this.socketFactory = socketFactory;
 
-    this.underlyingSocket = new AtomicReference<>(socketFactory.createSocket(host, port));
-    this.underlyingSocket.get().setSoTimeout(SERVER_READ_TIMEOUT_MILLIS);
+    this.underlyingSocket = new AtomicReference<>(createAndConnectSocket());
```

Comment:
this is probably the original behavior but can be just connect on use instead of on constructor? it's not typical practice to do this (but instead connect on demand or have connect/disconnect methods).

Reply:
`ProxyConnectionHandler` creates an instance of `ReconnectingSocket` on use, so in essence we do connect on use. Updated the javadoc to prevent confusion.